### DISCUSSION TITLE
Consume API Routes from getStaticProps

### DISCUSSION THREAD

**Discussion by HaNdTriX** [vercel/vercel#3937] [General] [answered]
Is it possible the consume an [internal api of a Next.js project](https://nextjs.org/docs/api-routes/introduction) (e.g.: `pages/api/post.js`) inside `getStaticProps`? If yes how?

**Example:**

```js
export async function getStaticProps(context) {
  // The following url must be absolute
  // but what is the correct origin/port? 
  const url = '/api/post'

  return {
    props: await fetch(url).then(res => res.json())
  }
}
```

Before (in `getInitialProps`) we had the `Request` object of node (`ctx.req`) that allowed us the retrieve this URL from some headers. 

Are the api lamdas running when the `getStaticProps` is beeing  executed the first time (on build)?

**Comment by paulogdm** [comment, marked as answer]
Yes it is, but you may need to do:
```
export async function getStaticProps(context) {
  // The following url must be absolute
  // but what is the correct origin/port? 
  const url = context.request.headers.host + '/api/post'

  return {
    props: await fetch(url).then(res => res.json())
  }
}
```
You need to access the "host" header in order to retrieve the information from where your request is coming from.
